cpp/src/parquet/encoding.cc

```diff
@@ -2479,7 +2481,18 @@ class DeltaBitPackDecoder : public DecoderImpl, virtual public TypedDecoder<DTyp
       if (ARROW_PREDICT_FALSE(values_current_mini_block_ == 0)) {
         if (ARROW_PREDICT_FALSE(!block_initialized_)) {
           buffer[i++] = last_value_;
-          if (ARROW_PREDICT_FALSE(i == max_values)) break;
+          if (ARROW_PREDICT_FALSE(i == max_values)) {
+            // When block is uninitialized and i reaches max_values we have two
+            // different possibilities:
+            // 1. i == total_value_count_, which means that the page may have only one
+            // value and we should not initialize any block.
+            // 2. i != total_value_count_ which means that user just read the first value
+            // in the page, so we should initialize the incoming block.
+            if (i != static_cast<int>(total_value_count_)) {
+              InitBlock();
+            }
```

Comment:
Wouldn't it be clearer to write this as:
```suggestion
            if (total_value_count_ != 1) {
              InitBlock();
            }
```

Reply:
By the way, should we add `ARROW_PREDICT_FALSE` here? @pitrou

Reply:
It's not useful since this case is exceptional already.